internal/ethapi/api_extra.go

```diff
@@ -115,7 +115,7 @@ func (s *BlockChainAPI) stateQueryBlockNumberAllowed(blockNumOrHash rpc.BlockNum
 		number = uint64(blockNumOrHash.BlockNumber.Int64())
 	} else {
 		block, err := s.b.BlockByNumberOrHash(context.Background(), blockNumOrHash)
```

Comment:
nit: I wonder if we should just use `BlockByHash` instead of having an invariant that this will always be called with a hash. (or really directly use `BlockByNumberOrHash` instead of checking `.BlockNumber !=nil)